presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java

```diff
@@ -61,9 +67,22 @@ else if ("Mac OS X".equals(osName)) {
             failRequirement("Presto requires a little endian platform (found %s)", ByteOrder.nativeOrder());
         }
 
+        verifyFileDescriptor();
+
         verifySlice();
     }
 
+    private static void verifyFileDescriptor()
+    {
+        OperatingSystemMXBean operatingSystemMXBean = ManagementFactory.getOperatingSystemMXBean();
```

Comment:
Make this a method lik `getFileDescriptorCount` and have it fail if it can't get the count. Use reflection to avoid failing to load if the class isn't found